util/test/account_testutil.go

```diff
@@ -211,6 +213,45 @@ func MakeCreateAppTxn(sender basics.Address) transactions.SignedTxnWithAD {
 	}
 }
 
+// MakeComplexCreateAppTxn makes a transaction that creates an arbitrary app
+func MakeComplexCreateAppTxn(sender basics.Address, approval, clear string, assemblerVersion ...uint64) (transactions.SignedTxnWithAD, error) {
```

Comment:
The vararg parameter to sneak in an optional parameter is confusing. I would prefer a comment saying "set assemblerVersion to 0 unless you want to override", or if you want to cleaner parameters, a second Make function which only has 3 parameters.

Reply:
I'll make the 4th parameter required and explain in a comment that 0 can be used for falling back to `AssemblerDefaultVersion`